tests/foreach/foreach.t

```diff
@@ -0,0 +1,88 @@
+SETUP TEST DIR
+  $ cp -r $TESTDIR/cram_dir/test_dir_1 test_dir_1
+  $ cd test_dir_1
+  $ git init
+  .* (re)
+  $ git remote add origin file://$TESTDIR/cram_dir/test_dir_1
+  $ git add *
+  $ git commit -m "First Commit"
+  .* (re)
+  .* (re)
+  .* (re)
+  $ mkdir src
+  $ cd src
+  $ cp -r $TESTDIR/cram_dir/test_dir_2 test_dir_2
+  $ cd test_dir_2
+  $ git init
+  .* (re)
+  $ git remote add origin file://$TESTDIR/cram_dir/test_dir_2
+  $ git add *
+  $ git commit -m "First Commit"
+  .* (re)
+  .* (re)
+  .* (re)
+  $ cd ..
+  $ cp -r $TESTDIR/cram_dir/test_dir_3 test_dir_3
+  $ cd test_dir_3
+  $ git init
+  .* (re)
+  $ git add *
+  $ git commit -m "First Commit"
+  .* (re)
+  .* (re)
+  .* (re)
+  $ git remote add origin file://$TESTDIR/cram_dir/test_dir_3
+  $ cd ..
+  $ cp -r $TESTDIR/cram_dir/test_dir_4 test_dir_4
+  $ cd test_dir_4
+  $ git init
+  .* (re)
+  $ git add *
+  $ git commit -m "First Commit"
+  .* (re)
+  .* (re)
+  .* (re)
+  $ git remote add origin file://$TESTDIR/cram_dir/test_dir_4
+  $ cd ../..
+  $ quark freeze > /dev/null
+
+TEST FOREACH WITH COMMAND
+  $ quark foreach 'echo $name $sm_path $displaypath $sha1 $toplevel $rev $version_control'
+  .* (re)
+  .* (re)
+  Entering test_dir_2
```

Comment:
Thinking of it, as of today the actual ordering of submodules is not contractual (and, depending on Python version, it will change, as the `dict` sort order before Python 3.7 is not deterministic, and even then the order used by quark to put stuff in is non-contractual); either you change the test to make sure it's ordering-insensitive (maybe pipe it in `sort`), or change the code of `quark foreach` so that the order of iteration is guaranteed. Given #24, probably the second option is better.

Reply:
Changed in https://github.com/comelz/quark/pull/27/commits/2ded4399a610137385ae00b59b739c89ece7a2b1